x-pack/filebeat/tests/integration/shipper_test.go

```diff
@@ -185,6 +212,61 @@ processors:
 		require.NoError(t, err, "error doing search request: %s", err)
 		return res.Hits.Total.Value == 2
 	}, 30*time.Second, 250*time.Millisecond, "never found 2 documents")
-	// ToDo add comparison of docs to make sure they are the same
-	// for example right now input.type is being overwritten with shipper
+
+	res, err := es.Search().
+		Index(".ds-filebeat-*").
+		Request(&search.Request{
+			Query: &types.Query{
+				Match: map[string]types.MatchQuery{
+					"message": {
+						Query:    uniqMsg,
+						Operator: &operator.And,
+					},
+				},
+			},
+		}).Do(context.Background())
+	require.Equal(t, int64(2), res.Hits.Total.Value)
+	diff, err := diffDocs(res.Hits.Hits[0].Source_,
+		res.Hits.Hits[1].Source_)
+	require.NoError(t, err, "error diffing docs")
+	if len(diff) != 0 {
+		t.Fatalf("docs differ:\n:%s\n", diff)
+	}
+}
+
+func diffDocs(doc1 json.RawMessage, doc2 json.RawMessage) (string, error) {
+	fieldsToDrop := []string{
+		"@timestamp",
+		"agent.ephemeral_id",
+		"agent.id",
+		"elastic_agent.id",
+	}
+	var d1 map[string]interface{}
+	var d2 map[string]interface{}
+
+	if err := json.Unmarshal(doc1, &d1); err != nil {
+		return "", err
+	}
+
+	if err := json.Unmarshal(doc2, &d2); err != nil {
+		return "", err
+	}
+	f1 := mapstr.M(d1).Flatten()
+	f2 := mapstr.M(d2).Flatten()
+
+	for _, key := range fieldsToDrop {
+		_ = f1.Delete(key)
+	}
+
+	for _, key := range fieldsToDrop {
+		_ = f2.Delete(key)
+	}
```

Comment:
It's just a matter of style, but that could be a single for loop:
```suggestion
	for _, key := range fieldsToDrop {
		_ = f1.Delete(key)
		_ = f2.Delete(key)
	}
```

Reply:
thanks, reduced to single loop